### PR TITLE
Add ReadScanline, WriteScanline and ScanlineSize.

### PR DESCRIPTION
**Pull request by JoostvanPinxten**
Fixes issue #101, by exposing the c-functions ReadScanline, WriteScanline and ScanlineSize.

### CODE REVIEW

**Comment by JoostvanPinxten** [comment]
Have I use the @debug decorator correctly? The other size functions did not have the @debug, while the other read and write functions did have it. So I applied that same approach here as well.

File: libtiff/libtiff_ctypes.py [hunk 1]
```diff
@@ -1210,6 +1210,20 @@ def NumberOfStrips(self):
         return libtiff.TIFFNumberOfStrips(self).value
     numberofstrips = NumberOfStrips
 
+    @debug
+    def WriteScanline(self, buf, row):
```
Comment (pearu) [inline libtiff/libtiff_ctypes.py]: Add the `sample=0` argument to the method and update the function call accordingly.

File: libtiff/libtiff_ctypes.py [hunk 2]
```diff
@@ -1210,6 +1210,20 @@ def NumberOfStrips(self):
...
+        return libtiff.TIFFWriteScanline(self, buf, row, 0)
+    writescanline = WriteScanline
+    
+    @debug
+    def ReadScanline(self, buf, row):
```
Comment (pearu) [inline libtiff/libtiff_ctypes.py]: Same note about `sample` as above.

File: libtiff/libtiff_ctypes.py (hunk 1 above)
Comment (JoostvanPinxten) [reply libtiff/libtiff_ctypes.py]: Good catch! I've updated accordingy; we've just never had a use case for a non-0 sample argument.

File: libtiff/libtiff_ctypes.py [hunk 3]
```diff
@@ -1211,13 +1211,13 @@ def NumberOfStrips(self):
...
-    def WriteScanline(self, buf, row):
-        return libtiff.TIFFWriteScanline(self, buf, row, 0)
+    def WriteScanline(self, buf, row, sample=0):
+        return libtiff.TIFFWriteScanline(self, buf, row, sample)
     writescanline = WriteScanline
     
     @debug
-    def ReadScanline(self, buf, row): 
-        return libtiff.TIFFReadScanline(self, buf, row, 0)
+    def ReadScanline(self, buf, sample=0):
```
Comment (pearu) [inline libtiff/libtiff_ctypes.py]: Looks like the row argument got lost..

**Pull request merged**